pkg/trace/stats/aggregation.go

```diff
@@ -64,8 +80,35 @@ func getStatusCode(s *pb.Span) uint32 {
 }
 
 // NewAggregationFromSpan creates a new aggregation from the provided span and env
-func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregationKey, enablePeerSvcAgg bool) Aggregation {
+func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregationKey, enablePeerSvcAgg bool, customTagConf []string, customTagSpanNames []string) Aggregation {
 	synthetics := strings.HasPrefix(origin, tagSynthetics)
+
+	i := 0
+
+	customKey := CustomTagKey("")
+
+	if len(customTagSpanNames) == 0 {
+		customTags := make([]string, len(customTagConf))
+		for k := range customTagConf {
+			customTags[i] = customTagConf[k]
+			i++
+		}
+
+		customKey = NewCustomTagKey(customTags)
+
+	} else {
+		for _, spanName := range customTagSpanNames {
+			if s.Name == spanName {
```

Comment:
Could we check a map instead of an array for faster look-ups?

Also, this check would happen before getting the custom tag key conf since we need to know the span name in order to know the associated custom tags.